frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -24,50 +34,80 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
 }) => {
   const { t } = useTranslation();
   const [handlePromise, inProgress, errorMessage] = usePromiseHandler();
-  const submit = React.useCallback(
-    (event: React.FormEvent<HTMLFormElement>): void => {
-      event.preventDefault();
-      const deleteOptions = {
-        kind: 'DeleteOptions',
-        apiVersion: 'v1',
-        propagationPolicy: 'Foreground',
-      };
-      handlePromise(
-        Promise.all([
-          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
-          ...(subscription?.status?.installedCSV
-            ? [
-                k8sKill(
-                  ClusterServiceVersionModel,
-                  {
-                    metadata: {
-                      name: subscription.status.installedCSV,
-                      namespace: subscription.metadata.namespace,
-                    },
-                  },
-                  {},
-                  deleteOptions,
-                ).catch(() => Promise.resolve()),
-              ]
-            : []),
-        ]),
-      )
-        .then(() => {
-          close();
-          if (
-            window.location.pathname.split('/').includes(subscription.metadata.name) ||
-            window.location.pathname.split('/').includes(subscription.status.installedCSV)
-          ) {
-            history.push(
-              resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()),
-            );
-          }
-        })
-        .catch(() => {});
-    },
-    [close, handlePromise, k8sKill, subscription],
+
+  const canPatchConsoleOperatorConfig = useAccessReview({
+    group: ConsoleOperatorConfigModel.apiGroup,
+    resource: ConsoleOperatorConfigModel.plural,
+    verb: 'patch',
+    name: CONSOLE_OPERATOR_CONFIG_NAME,
+  });
+
+  const csvPlugins = getClusterServiceVersionPlugins(csv?.metadata?.annotations);
+
+  const [consoleOperatorConfig] = useK8sWatchResource<K8sResourceKind>(
+    canPatchConsoleOperatorConfig && csvPlugins.length > 0
+      ? {
+          kind: referenceForModel(ConsoleOperatorConfigModel),
+          isList: false,
+          name: CONSOLE_OPERATOR_CONFIG_NAME,
+        }
+      : null,
+  );
+
+  const enabledPlugins = csvPlugins.filter((plugin) =>
+    isPluginEnabled(consoleOperatorConfig, plugin),
   );
 
+  const removePlugins: boolean =
+    !!consoleOperatorConfig && canPatchConsoleOperatorConfig && enabledPlugins.length > 0;
+
+  const submit = (event: React.FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    const deleteOptions = {
+      kind: 'DeleteOptions',
+      apiVersion: 'v1',
+      propagationPolicy: 'Foreground',
+    };
+
+    const patch =
+      removePlugins &&
+      enabledPlugins.map((plugin) => getPluginPatch(consoleOperatorConfig, plugin, false));
```

Comment:
We should make sure `patch` is only ever assigned an object, not a boolean.

```suggestion
      removePlugins ?
      enabledPlugins.map((plugin) => getPluginPatch(consoleOperatorConfig, plugin, false)) : null;
```